cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -226,22 +175,21 @@ func tpmSealKey(key partition.EncryptionKey, rkey partition.RecoveryKey, options
 		fmt.Sprintf("snapd_recovery_mode=recover snapd_recovery_system=%s console=ttyS0 console=tty1 panic=-1", options.SystemLabel),
 	}
 
-	tpm.SetKernelCmdlines(kernelCmdlines)
-
-	if options.Model != nil {
-		tpm.SetModels([]*asserts.Model{options.Model})
-	}
-
-	// Provision the TPM as late as possible
-	// TODO:UC20: ideally we should ask the firmware to clear the TPM and then reboot
-	//            if the device has previously been provisioned, see
-	//            https://godoc.org/github.com/snapcore/secboot#RequestTPMClearUsingPPI
-	if err := tpm.Provision(); err != nil {
-		return fmt.Errorf("cannot provision the TPM: %v", err)
+	sealKeyParams := secboot.SealKeyParams{
+		ModelParams: []*secboot.SealKeyModelParams{
+			{
+				Model:          options.Model,
+				KernelCmdlines: kernelCmdlines,
+				EFILoadChains:  [][]string{loadChain},
```

Comment:
The original idea was to allow using different chains to allow rollbacks in a reseal scenario such as key rotation, so that if a kernel signed with a new key fails to boot, it could still go back and boot the previous kernel. (This would also require two reseals, one with both versions and another with the latest one once it boots correctly. But the exact steps to follow are not defined yet.)

Reply:
(Also we may have different chains for modes under the same model.)